velox/functions/sparksql/DecimalVectorFunctions.cpp

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+// Rescale two inputs as the same scale and compare. Returns 0 when a is equal
+// with b. Returns -1 when a is less than b. Returns 1 when a is greater than b.
+template <typename T>
+int32_t rescaleAndCompare(T a, T b, int32_t deltaScale) {
+  T aScaled = a;
+  T bScaled = b;
+  if (deltaScale < 0) {
+    aScaled = a * velox::DecimalUtil::kPowersOfTen[-deltaScale];
+  } else if (deltaScale > 0) {
+    bScaled = b * velox::DecimalUtil::kPowersOfTen[deltaScale];
+  }
+  if (aScaled == bScaled) {
+    return 0;
+  } else if (aScaled < bScaled) {
+    return -1;
+  } else {
+    return 1;
+  }
+}
+
+// Compare two decimals. Rescale one of them When they are of different scales.
+int32_t decimalCompare(
+    int128_t a,
+    uint8_t aPrecision,
+    uint8_t aScale,
+    int128_t b,
+    uint8_t bPrecision,
+    uint8_t bScale) {
+  int32_t deltaScale = aScale - bScale;
+  // Check if we need 256-bits after adjusting the scale.
+  bool need256 = (deltaScale < 0 &&
+                  aPrecision - deltaScale > LongDecimalType::kMaxPrecision) ||
+      (bPrecision + deltaScale > LongDecimalType::kMaxPrecision);
+  if (need256) {
+    return rescaleAndCompare<int256_t>(
+        static_cast<int256_t>(a), static_cast<int256_t>(b), deltaScale);
+  }
+  return rescaleAndCompare<int128_t>(a, b, deltaScale);
+}
+
+// GreaterThan decimal compare function.
```

Comment:
This comment is redundant as it simply repeats the code. Please, remove. Ditto other places.